src/library_syscall.js

```diff
@@ -1175,7 +1175,7 @@ var SyscallsLibrary = {
   },
   __sys_fstatfs64: function(fd, size, buf) {
     var stream = SYSCALLS.getStreamFromFD(fd);
-    return ___sys_statfs64(0, size, buf);
+    return __sys_statfs64(stream.path, size, buf);
```

Comment:
The extra `_` is needed, as compiled code gets a `_` prefix when put into the JS global namespace. However, it may be undefined, as you noticed - we need to add a dependency here. This will do that:
```diff
diff --git a/src/library_syscall.js b/src/library_syscall.js
index e9eb9134b..9197ff2c1 100644
--- a/src/library_syscall.js
+++ b/src/library_syscall.js
@@ -1173,6 +1173,7 @@ var SyscallsLibrary = {
     {{{ makeSetValue('buf', C_STRUCTS.statfs.f_namelen, '255', 'i32') }}};
     return 0;
   },
+  __sys_fstatfs64__deps: ['__sys_statfs64'],
   __sys_fstatfs64: function(fd, size, buf) {
     var stream = SYSCALLS.getStreamFromFD(fd);
     return ___sys_statfs64(0, size, buf);
```
The second change, `stream.path`, looks good to me!

To verify both, please add a testcase. Perhaps adding to `tests/core/test_statvfs.c` would be simplest.